docs/build/isc/v1.3/docs/tutorials/defi-lend-borrow-tutorial-part-1.md

```diff
@@ -0,0 +1,289 @@
+    - Updates the total amount borrowed in the contract.
+    - Transfers the borrowed amount of the underlying token to the user.
+    - Returns a boolean indicating whether the borrowing process was successful.
+
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/IToken.sol#L166-L196
+
+```
+- Repay: This method allows users to repay the borrowed underlying token.
+  - Ensures the repayment amount does not exceed the user's borrow balance.
+  - Transfers the repayment amount from the user to the contract.
+  - Calculates the interest on the repayment amount and adjusts the user's borrow balance accordingly.
+  - Updates the total amount borrowed and the contract's reserves.
+  - Updates the user's collateral in the `ITokenManager`.
+  - Returns a boolean indicating whether the repayment process was successful.
+
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/IToken.sol#L198-L227
+
+```
+- `getBorrowRate()`: Returns the current borrow rate per block.
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/IToken.sol#L229-L236
+
+```
+- `getSupplyRate()`: Returns the current supply rate per block.
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/IToken.sol#L238-L245
+
+```
+You can review the full code for the [IToken contract in the tutorial's Repository](https://github.com/iota-community/Defi-lend-borrow/blob/main/contracts/IToken.sol).
+
+### InterestRateModel
+
+The `InterestRateModel` contract calculates the interest rates for borrowing and supplying assets based on the utilization of the underlying assets.
+
+#### Key Functions
+- `utilizationRate(uint cash, uint borrows, uint reserves)`: Calculates the utilization rate of the market.
+
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/InterestRateModel.sol#L41-L57
+
+```
+- `getBorrowRate(uint cash, uint borrows, uint reserves)`: Calculates the current borrow rate per block.
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/InterestRateModel.sol#L59-L68
+
+```
+- `getSupplyRate(uint cash, uint borrows, uint reserves, uint reserveFactorMantissa)`: Calculates the current supply rate per block.
+```solidity reference
+https://github.com/iota-community/Defi-lend-borrow/blob/3a368391f4767d1decb209ad6bfdd20a2b08fa03/contracts/InterestRateModel.sol#L70-L79
+
+``` 
+
+We have now covered all relevant parts and working of the InterestRateModal contract, here is the [link](https://github.com/iota-community/Defi-lend-borrow/blob/main/contracts/InterestRateModel.sol) for the full contract code for your reference.
```

Comment:
please don't create links on words like `link`, `here`, and so on. Instead, use a more descriptive term, for example, You can find the full contract code in the Defi-lend-borrow repository